src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2068,6 +2068,79 @@ void ApplicationManagerImpl::ProcessServiceStatusUpdate(
       service_type, service_event, reason, app_id);
 
   rpc_service_->ManageHMICommand(notification);
+
+  if (hmi_apis::Common_ServiceEvent::REQUEST_REJECTED == service_event &&
+      StopServicesForApp(app, service_type)) {
+    state_ctrl_.SetRegularState(app,
+                                mobile_apis::PredefinedWindows::DEFAULT_WINDOW,
+                                mobile_apis::HMILevel::HMI_NONE,
+                                true);
+  }
+}
+
+bool ApplicationManagerImpl::StopServicesForApp(
+    ApplicationSharedPtr app,
+    const hmi_apis::Common_ServiceType::eType service_type) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  bool result = true;
+  if (!app) {
+    LOG4CXX_WARN(logger_, "Received invalid app");
+    return false;
+  }
+
+  switch (service_type) {
+    case hmi_apis::Common_ServiceType::VIDEO: {
+      {
+        sync_primitives::AutoLock lock(navi_service_status_lock_);
+        auto app_services = navi_service_status_.find(app->app_id());
+        if (navi_service_status_.end() != app_services) {
+          navi_service_status_.erase(app_services);
```

Comment:
Deleting the item from navi_service_status_ will delete the status for both video and audio, even though Video is the only service type specified here. EndNaviServices has a cleaner way of addressing this issue.